src/octree/batch_iterator.rs

```diff
@@ -1,31 +1,47 @@
 use crate::errors::*;
-use crate::math::{Isometry3, Obb, OrientedBeam};
+use crate::math::PointCulling;
+use crate::math::{AllPoints, Isometry3, Obb, OrientedBeam};
 use crate::octree::{self, Octree};
 use crate::{LayerData, Point, PointData};
-use cgmath::{Decomposed, Matrix4, Vector3, Vector4};
+use cgmath::{Matrix4, Vector3, Vector4};
 use collision::Aabb3;
 use fnv::FnvHashMap;
 
 /// size for batch
 pub const NUM_POINTS_PER_BATCH: usize = 500_000;
 
-///possible kind of iterators that can be evaluated in batch of points in BatchIterator
 #[allow(clippy::large_enum_variant)]
-#[derive(Clone)]
-pub enum PointCulling {
-    Any(),
+#[derive(Debug, Clone)]
+pub enum GeoFence {
```

Comment:
I think it might be clearer if we name this PointLocation, and what's currently called PointLocation can become PointQuery. GeoFence has a specific meaning: https://en.wikipedia.org/wiki/Geo-fence

Reply:
good idea, I was running short on names!

Reply:
done!